packages/compass-collection/src/modules/stats.ts

```diff
@@ -112,16 +125,21 @@ export const updateCollectionDetails = (
  *
  * @returns {any} The new state.
  */
-const reducer = (
-  state = getInitialState(),
-  action: AnyAction
-): CollectionStatsObject => {
-  if (action.type === UPDATE_COLLECTION_DETAILS) {
-    return action.stats;
-  } else if (action.type === RESET_COLLECTION_DETAILS) {
-    return getInitialState();
+const reducer = (state = getInitialState(), action: AnyAction): State => {
+  switch (action.type) {
+    case ActionTypes.UpdateCollectionDetails:
+      return {
+        ...state,
+        [action.namespace]: action.stats,
```

Comment:
We are always adding new stats with this action, but never cleaning up unused ones (when tabs are closed or collection gets deleted), probably should clean up so we don't keep stale copies around?

Reply:
Updated it to clean up when tabs are closed and no longer active 👍